Bio/SeqUtils/__init__.py

```diff
@@ -667,6 +667,40 @@ def calculate(self, sequence):
 
         return exp(cai_value / cai_length)
 
+    def optimize(self, sequence, seq_type="DNA", table=standard_dna_table):
+        """Return a new DNA sequence with preferred codons only."""
+        try:  # If seq record is provided, convert to sequence
+            name = sequence.id
+            sequence = sequence.seq
+        except AttributeError:  # not a  SeqRecord object
+            pass
+        seq = sequence.upper()
+        # Make dict with amino acids referencing preferred codons
+        pref_codons = {aminoacid: [] for aminoacid in table.protein_alphabet}
+        for codon, aminoacid in table.forward_table.items():
+            if self[codon] == 1.0:
+                pref_codons[aminoacid] = codon
+        # Create amino acid sequence if DNA was provided
+        if seq_type == "DNA":
+            aa_seq = translate(seq)
+        elif seq_type == "protein":
+            aa_seq = seq
+        else:
+            raise TypeError
```

Comment:
Is it not meaningful to support RNA?

Reply:
I didn't imagine a case where I would want to use RNA, but it's possible someone would. 
DNA and RNA would be handled the same way by translate(), though; I've update the code to accept seq_type = "RNA".

Reply:
Looking at this again, two minor points.

First can you add an exception message about the argument not being a supported sequence type.

And second, maybe ValueError not TypeError?